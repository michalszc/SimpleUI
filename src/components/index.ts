export * from "./Button";
export * from "./ColorPicker";
export * from './Switch';
export * from './FormLabel';