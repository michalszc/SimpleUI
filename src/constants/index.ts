export { 
    default as Colors,
    red,
    pink,
    purple,
    deepPurple,
    indigo,
    blue,
    lightBlue,
    cyan,
    teal,
    green,
    lightGreen,
    lime,
    yellow,
    amber,
    orange,
    deepOrange,
    brown,
    grey,
    blueGrey,
    black,
    white,
    transparent
} from './Colors';
export type { Color, IColors } from './Colors';
export { default as Palette } from './Palette';
export type { IPalette, PaletteColor } from './Palette';
export { default as Typography } from './Typography';
export type { ITypography, TypographyVariant } from './Typography';
export { default as Breakpoints } from './Breakpoints';
export type { IBreakpoints } from './Breakpoints';
export type { AbsoluteUnits, RelativeUnits, Units } from './units';
export { default as Sizes } from './Sizes';
export type { ISizes } from './Sizes';
