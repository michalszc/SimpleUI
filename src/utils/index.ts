export * from "./types";
export type { StyleProps } from "./styles";
export * from './theme';
export * from './convertColors';