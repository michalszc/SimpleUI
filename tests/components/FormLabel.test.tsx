import React from "react";
import '@testing-library/jest-dom';
import { fireEvent, render } from "@testing-library/react";
import { FormLabel, Switch } from "../../src"; 

describe("FormLabel", () => {
    test("should properly render FormLabel component", () => {
        const container = render(<FormLabel />);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Header with children", () => {
        const { getByText } = render(
            <FormLabel>
                test
            </FormLabel>
        );
        const label = getByText('test');
        expect(label).toBeTruthy();
    });
    test("should properly handle htmlfor attribute", () => {
        const mockFn = jest.fn();
        const { getByText } = render(
            <>
                <FormLabel htmlFor={'formlabel-test-id'}>
                    test
                </FormLabel>
                <Switch onChange={mockFn} id={'formlabel-test-id'}/>
            </>
        );
        const label = getByText('test');
        expect(label).toBeTruthy();
        fireEvent.click(label);
        expect(mockFn).toBeCalled();
    });
});
