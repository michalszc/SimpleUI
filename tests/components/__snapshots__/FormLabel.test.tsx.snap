// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel should properly render FormLabel component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-bjfHbI bnjIFt"
        id="4fzzzxjylrx"
        mode="light"
      />
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-bjfHbI bnjIFt"
      id="4fzzzxjylrx"
      mode="light"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
